### PR TITLE
D8 nid 1678

### PR DESCRIPTION
**Pull request by omahm**
Replace telephone field with phone field and copy telephone data across. Update all config & code references to field_telephone.

### CODE REVIEW

File: config/sync/core.entity_view_display.node.gp_practice.default.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@ dependencies:
     - field.field.node.gp_practice.field_gp_surgery_name
     - field.field.node.gp_practice.field_location
     - field.field.node.gp_practice.field_meta_tags
+    - field.field.node.gp_practice.field_phone
     - field.field.node.gp_practice.field_telephone
```
Comment (johangant) [inline config/sync/core.entity_view_display.node.gp_practice.default.yml:20]: field_telephone is still listed here as a dependency, although it's replaced by field_phone across the board. Can we take this out of the list of config dependencies?

File: config/sync/core.entity_view_display.node.gp_practice.diff.yml [hunk 2]
```diff
@@ -17,6 +17,7 @@ dependencies:
     - field.field.node.gp_practice.field_gp_surgery_name
     - field.field.node.gp_practice.field_location
     - field.field.node.gp_practice.field_meta_tags
+    - field.field.node.gp_practice.field_phone
     - field.field.node.gp_practice.field_telephone
```
Comment (johangant) [inline config/sync/core.entity_view_display.node.gp_practice.diff.yml:21]: Another mention of field_telephone as a config dependency

File: config/sync/core.entity_view_display.node.gp_practice.diff.yml [hunk 3]
```diff
@@ -511,6 +513,7 @@ hidden:
   field_contact_town_city: true
   field_gp_practice_name: true
   field_gp_surgery_name: true
+  field_telephone: true
```
Comment (johangant) [inline config/sync/core.entity_view_display.node.gp_practice.diff.yml:516]: Should this be field_phone?

File: config/sync/core.entity_view_display.node.gp_practice.full.yml [hunk 4]
```diff
@@ -17,6 +17,7 @@ dependencies:
     - field.field.node.gp_practice.field_gp_surgery_name
     - field.field.node.gp_practice.field_location
     - field.field.node.gp_practice.field_meta_tags
+    - field.field.node.gp_practice.field_phone
     - field.field.node.gp_practice.field_telephone
```
Comment (johangant) [inline config/sync/core.entity_view_display.node.gp_practice.full.yml:21]: Remove field_telephone?

File: config/sync/core.entity_view_display.node.gp_practice.full.yml [hunk 5]
```diff
@@ -458,6 +460,7 @@ hidden:
   field_gp_practice_no: true
   field_gp_surgery_name: true
   field_meta_tags: true
+  field_telephone: true
```
Comment (johangant) [inline config/sync/core.entity_view_display.node.gp_practice.full.yml:463]: Should this be field_phone?

File: config/sync/core.entity_view_display.node.gp_practice.search_result.yml [hunk 6]
```diff
@@ -17,6 +17,7 @@ dependencies:
     - field.field.node.gp_practice.field_gp_surgery_name
     - field.field.node.gp_practice.field_location
     - field.field.node.gp_practice.field_meta_tags
+    - field.field.node.gp_practice.field_phone
     - field.field.node.gp_practice.field_telephone
```
Comment (johangant) [inline config/sync/core.entity_view_display.node.gp_practice.search_result.yml:21]: Remove field_telephone?

File: config/sync/core.entity_view_display.node.gp_practice.search_result.yml [hunk 7]
```diff
@@ -94,6 +95,7 @@ hidden:
   field_gp_surgery_name: true
   field_location: true
   field_meta_tags: true
+  field_phone: true
   field_telephone: true
```
Comment (johangant) [inline config/sync/core.entity_view_display.node.gp_practice.search_result.yml:99]: Remove field_telephone?

File: config/sync/core.entity_view_display.node.gp_practice.teaser.yml [hunk 8]
```diff
@@ -17,6 +17,7 @@ dependencies:
     - field.field.node.gp_practice.field_gp_surgery_name
     - field.field.node.gp_practice.field_location
     - field.field.node.gp_practice.field_meta_tags
+    - field.field.node.gp_practice.field_phone
     - field.field.node.gp_practice.field_telephone
```
Comment (johangant) [inline config/sync/core.entity_view_display.node.gp_practice.teaser.yml:21]: Remove field_telephone?

File: config/sync/core.entity_view_display.node.gp_practice.teaser.yml [hunk 9]
```diff
@@ -85,6 +86,7 @@ hidden:
   field_gp_surgery_name: true
   field_location: true
   field_meta_tags: true
+  field_phone: true
   field_telephone: true
```
Comment (johangant) [inline config/sync/core.entity_view_display.node.gp_practice.teaser.yml:90]: Remove field_telephone?

File: web/modules/custom/nidirect_common/nidirect_common.install [hunk 10]
```diff
@@ -322,6 +322,25 @@ function nidirect_common_update_8005() {
 }
 
 
+/**
+ * Copy telephone field values to phone field to fix issue raise in D8NID-1678.
+ */
+function nidirect_common_update_8006() {
+  $database = \Drupal::database();
+
+  // Use 'IGNORE' to prevent 'Duplicate entry' warnings.
```
Comment (johangant) [inline web/modules/custom/nidirect_common/nidirect_common.install:331]: Nice, this looks a good bit simpler than using some kind of intermediary field as storage

**Review by johangant — APPROVED** [review]
Some questions about whether field_telephone has remained in some of the config dependencies but otherwise looks good, so preemptively approving

**Comment by omahm** [comment]
Keeping all the telephone field stuff in there, later on when we delete the field completely all of that config will go as well.

**Pull request merged**